datafusion/src/physical_plan/functions.rs

```diff
@@ -543,6 +553,28 @@ macro_rules! invoke_if_unicode_expressions_feature_flag {
     };
 }
 
+///Function volatility determines when a function can be inlined or in the future have evaluations elided when the arguments are the same
+///Immutable - a pure function which always remains the same
+///Stable - Maybe not required?. Used for functions which can modify the db but are stable for multiple calls in statement
+///Volatile - Functions where output can vary for each call.
+///For more information see https://www.postgresql.org/docs/current/xfunc-volatility.html
+#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[repr(u32)]
```

Comment:
why this representation?

Reply:
No particular reason I just wanted to be able to filter on the volatility being equal to or stricter than a given level and I wasn't sure how the derived derived implementations would do it. I'll remove the representation and switch to the default derived version.